lib/infrastructure/dex_token.repository.dart

```diff
@@ -49,100 +53,96 @@ class DexTokenRepositoryImpl with ModelParser implements DexTokenRepository {
   Future<List<DexToken>> getTokensFromAccount(
     String accountAddress,
   ) async {
-    final dexTokens = <DexToken>[];
-    final balanceMap =
-        await aedappfm.sl.get<ApiService>().fetchBalance([accountAddress]);
+    final balanceMap = await apiService.fetchBalance([accountAddress]);
     final balance = balanceMap[accountAddress];
     if (balance == null) {
       return [];
     }
+    final dexTokens = <DexToken>[];
 
     final tokenAddressList = <String>[];
     for (final token in balance.token) {
       if (token.tokenId == 0) tokenAddressList.add(token.address!);
     }
 
-    final dexTokenUCO = ucoToken.copyWith(
-      balance: fromBigInt(balance.uco).toDouble(),
-      icon: 'Archethic.svg',
+    final dexTokenUCO = DexToken.uco(
+      balance: archethic.fromBigInt(balance.uco).toDouble(),
     );
     dexTokens.add(dexTokenUCO);
 
-    final tokenMap = await aedappfm.sl.get<ApiService>().getToken(
-          tokenAddressList,
-          request: 'name, symbol, properties',
-        );
+    final tokenMap = await apiService.getToken(
+      tokenAddressList,
+      request: 'name, symbol, properties',
+    );
 
+    // Fetch all token of pairs from API
+    final tokensAddresses = tokenMap.entries
+        .expand<String>((entry) {
+          final token1Address = entry.value.properties['token1_address'];
+          final token2Address = entry.value.properties['token2_address'];
+          if (token1Address == null || token2Address == null) return [];
+
+          return [
+            token1Address,
+            token1Address,
+          ];
+        })
+        .whereNot((address) => address == kUCOAddress)
+        .toList();
+
+    final tokensSymbolMap = await apiService.getToken(
+      tokensAddresses,
```

Comment:
Warning. we should remove doublons in `tokenAddresses`
Currently for example, when tokenMap contains LP Token, we have in properties `token1_adress`and `token2_adress` but LP token can have same token address in the pair

I try with LP token with aeETH (token1) and UCO (token2) and the result in `tokensAddresses` is the aeETH address twice